streamparse/util.py

```diff
@@ -259,6 +259,32 @@ def get_nimbus_client(env_config=None, host=None, port=None, timeout=7000):
                                 timeout=timeout)
     return nimbus_client
 
+_storm_workers = {}
+def get_storm_workers(env_config):
+    """Retrieves list of workers, optionally from nimbus
+
+    This function will look up the list of current workers from nimbus unless
+    `allow_worker_lookup: False` is set in `env_config` or if a list of workers
+    has been provided in `env_config`.
+    :param env_config: The project's parsed config.
+    :type env_config: `dict`
+
+    :returns: List of workers
+    """
+    nimbus_info = get_nimbus_host_port(env_config)
+    if _storm_workers.get(nimbus_info):
```

Comment:
Because of the weird case where an empty list is actually the correct, cached item, we should probably have this be `if nimbus_info in _storm_workers:`